Extension/test/scenarios/SingleRootProject/tests/lmTool.test.ts

```diff
@@ -0,0 +1,366 @@
+                targetPlatform: 'windows',
+                targetArchitecture: 'x64',
+                fileContext: {
+                    compilerArguments: compilerArguments
+                }
+            }
+        });
+
+        const result = await getProjectContext(mockTextDocumentStub.uri, context, new vscode.CancellationTokenSource().token);
+
+        ok(telemetryStub.calledOnce, 'Telemetry should be called once');
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            "language": 'C++'
+        })));
+        if (expectedCompiler) {
+            ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+                "compiler": expectedCompiler
+            })));
+        }
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            "standardVersion": 'C++20'
+        })));
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            "targetPlatform": 'Windows'
+        })));
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            "targetArchitecture": 'x64'
+        })));
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            "compilerArgumentCount": compilerArguments.length.toString()
+        })));
+        if (expectedCompilerArguments.length > 0) {
+            ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+                "filteredCompilerArguments": expectedCompilerArguments.join(' ')
+            })));
+        }
+        ok(telemetryStub.calledWithMatch('ProjectContext', sinon.match({
+            'time': sinon.match.string
+        })));
+        ok(result, 'result should not be undefined');
+        ok(result.language === 'C++');
+        ok(result.compiler === expectedCompiler);
+        ok(result.standardVersion === 'C++20');
+        ok(result.targetPlatform === 'Windows');
+        ok(result.targetArchitecture === 'x64');
+        ok(JSON.stringify(result.compilerArguments) === JSON.stringify(expectedCompilerArguments));
+    };
+
+    it('should provide compilerArguments based on copilotcppMsvcCompilerArgumentFilter.', async () => {
+        await testGetProjectContext({
+            compiler: 'msvc',
+            expectedCompiler: 'MSVC',
+            context: { flags: { copilotcppMsvcCompilerArgumentFilter: '{"^(/GR.*|/EH.*)$": ""}' } },
+            compilerArguments: ['/RTC1', '/GR-', '/Gd', '/MDd', '/GR', '/FC', '/W3', '/EHsc', '/Zc:wchar_t', '/ZI', '/EHs-'],
+            expectedCompilerArguments: ['/GR-', '/GR', '/EHsc', '/EHs-']
+        });
+    });
+
+    it('should provide compilerArguments based on copilotcppClangCompilerArgumentFilter.', async () => {
+        await testGetProjectContext({
+            compiler: 'clang',
+            expectedCompiler: 'Clang',
+            context: { flags: { copilotcppClangCompilerArgumentFilter: '{"^(-fno-exceptions|-fno-rtti|/GR-)$": "", "^-std=.*$": ""}' } },
```

Comment:
FYI, you shouldn't ever need to have -std in the filter, because it's already provided to Copilot by the standardVersion context.